multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -87,7 +87,13 @@
             }
 
             try (InputStream in = item.openStream()) {
-                FileEntry entry = fileService.addFile(spaceGuid, namespace, item.getName(), in);
+                FileEntry entry;
+                if (request.getHeader("X-File-Size") != null) {
+                    entry = fileService.addFile(spaceGuid, namespace, item.getName(), in,
+                                                Long.parseLong(request.getHeader("X-File-Size")));
```

Comment:
What if some moron passes a non-integer here? Will this be handled gracefully by our exception handlers?